src/server/HelixImportProject.js

```diff
@@ -28,6 +29,29 @@ export class HelixImportProject extends BaseProject {
     return this._allowInsecure;
   }
 
+  withHeadersFile(value) {
+    this._headersFile = value;
+    if (value) {
+      fs.readJSON(value).then((result) => {
+        this._cliHeaders = result;
+      }).catch((err) => {
+        this.log.error('error reading the headers file', err);
+        this._cliHeaders = {};
+      });
+    } else {
+      this._cliHeaders = {};
+    }
+    return this;
+  }
```

Comment:
this won't work. you can't guarantee when the file has finished reading and the _cliHedaers is used before the callback is invoked.

you need to read the file during `init()`

Reply:
which `init()`? I don't see an `init()` in this file. Do you mean in the `async start()`?

Reply:
yes. sorry, `start()`.

Reply:
I fixed the async issues when reading the headers file: https://github.com/adobe/helix-cli/pull/2368/commits/d1ebfc089d965608dae64b67728d5bbc924d5251
Also updated and added more tests.